File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> hmap = new Dictionary<int, int>();
        for(int i=0; i < nums.Length; i++){
            if(hmap.ContainsKey(nums[i])){
                return new int[] {i, hmap[nums[i]]};
            }
            else if (hmap.ContainsKey(target - nums[i])){ hmap[target - nums[i]] = i; }
            else hmap.Add(target - nums[i], i);
        }
        
        return null;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hmap = {}
        length = len(nums)
        for i in range(length):
            if nums[i] in hmap:
                return [i, hmap[nums[i]]]
            hmap[target - nums[i]] = i